src/heretic/p_user.c

```diff
@@ -560,6 +560,19 @@ void P_PlayerThink(player_t * player)
         ultimatemsg = false;    // clear out any chat messages.
         BorderTopRefresh = true;
     }
+    
+    // [crispy] Decrement centered message tics if greater than 0
+    if (player->centerMessageTics > 0)
+    {
+        player->centerMessageTics--;
+    }
+
+    if (!player->centerMessageTics)
```

Comment:
This code will have to get moved into the block above. Else, we would set `BorderTopRefresh = true` during each tic when `player->centerMessageTics` is zero, whereas we only want to set it once when `player->centerMessageTics` has just been decreased to zero.

Apart from this, all is fine!